packages/devtools_shared/lib/src/extensions/extension_model.dart

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Describes an extension that can be dynamically loaded into a custom screen
+/// in DevTools.
+class DevToolsExtensionConfig {
+  DevToolsExtensionConfig._({
+    required this.name,
+    required this.path,
+    required this.issueTrackerLink,
+    required this.version,
+    required this.materialIconCodePoint,
+  });
+
+  factory DevToolsExtensionConfig.parse(Map<String, Object?> json) {
+    // Defaults to the code point for [Icons.extensions_outlined] if null.
+    late int codePoint;
+    final codePointFromJson = json[materialIconCodePointKey];
+    const defaultCodePoint = 0xf03f;
+    if (codePointFromJson is String?) {
+      codePoint =
+          int.tryParse(codePointFromJson ?? '0xf03f') ?? defaultCodePoint;
+    } else {
+      codePoint = codePointFromJson as int? ?? defaultCodePoint;
+    }
+
+    final name = json[nameKey] as String?;
```

Comment:
We should use pattern matching here for JSON validation:

```dart
  if (json
      case {
        nameKey: String name,
        pathKey: String path,
        issueTrackerKey: String issueTracker,
        versionKey: String version,
      }) {
    return DevToolsExtensionConfig._(
      name: name,
      path: path,
      issueTrackerLink: issueTrackerLink,
      version: version,
      materialIconCodePoint: codePoint,
    );
  }
  throw StateError(
    'missing required fields ${nullFields.toString()} in the extension '
    'config.json',
  );
```

Reply:
I will address this comment as part of https://github.com/flutter/devtools/pull/6097.